### PR TITLE
[ENH] Implement finite BOLD response- separate (FS) modeling

### PR DESCRIPTION
**Pull request by tsalo**
## Summary

Fixes None. References #197. 

Here is an example FS design matrix. 
fs_design_matrix

## List of changes proposed in this PR (pull-request)

- Add the `--fir-delays` argument to the CLI and all associated functions.
- Compute a separate beta series for each delay when using the FIR HRF model and the LSS estimator. This method is not applicable with the LSA estimator.

### CODE REVIEW

**Comment by tsalo** [comment]
@jdkent I still need to add tests and update documentation (hence the WIP label), but I wanted to get your thoughts on the method and code before doing all of that.

**Comment by jdkent** [comment]
Hi @tsalo, I'm currently working on reducing a backlog of duties, but I will get to this by next Tuesday at the latest.

**Comment by jdkent** [comment]
Hi @tsalo, sorry I haven't gotten to this yet, I've been working on simulations (using the LSA code you provided, thank you!) and I am presenting on that data next Tuesday. The process has been a little more taxing than I anticipated.

File: src/nibetaseries/interfaces/nistats.py [hunk 1]
```diff
@@ -86,15 +89,28 @@ def _run_interface(self, runtime):
...
+                    if new_delay_ttype in beta_maps:
+                        beta_maps[new_delay_ttype].append(beta_map)
+                    else:
+                        beta_maps[new_delay_ttype] = [beta_map]
             else:
-                beta_maps[trial_type] = [beta_map]
+                # calculate the beta map
+                beta_map = model.compute_contrast(trial_type, output_type='effect_size')
+                design_matrix_collector[trial_idx] = model.design_matrices_[0]
+                # import pdb; pdb.set_trace()
```
Comment (jdkent) [inline src/nibetaseries/interfaces/nistats.py]: Since we are touching this code anyways, we can remove this comment.

**Comment by tsalo** [comment]
I don't think FIR + LSA would work, because any overlapping timepoints across trials would have duplicate FIR regressors. I haven't tried it though, so maybe it's possible if the delay times are chosen carefully? Still, it's not something I've come across in the literature.

**Comment by tsalo** [comment]
I'm not sure why CI is currently breaking. My latest commit only removed the commented-out line of code and the commit before that was passing.

**Comment by jdkent** [comment]
This may be related to bids-standard/pybids#496, and it was fixed in niworkflows with poldracklab/niworkflows#405, but I'll see if there is something else going on (either instead or in addition to)

**Comment by jdkent** [comment]
Once #212 passes and I merge with master, this branch should be fixed once it is merged with master.

File: src/nibetaseries/cli/run.py [hunk 2]
```diff
@@ -82,6 +82,9 @@ def get_parser():
                                     'spm + derivative + dispersion'],
                            help='convolve your regressors '
                                 'with one of the following hemodynamic response functions')
+    proc_opts.add_argument('--fir-delays', default=None,
```
Comment (jdkent) [inline src/nibetaseries/cli/run.py]: Inside `main`, we should add a check to make sure that if `opts.hrf_model ==  'fir'` then `opts.fir_delays` should not be `None`.

Comment (jdkent) [reply src/nibetaseries/cli/run.py]: or you can follow the strategy I used [here](https://github.com/HBClab/NiBetaSeries/pull/213/files#diff-177742e0b69cb7dc3d6ba7782e9a2293R60) to make the command line arguments conditionally dependent.

**Comment by jdkent** [comment]
P.S. I'm adding you as a collaborator, so when you have an approving review, you can merge in the pull request yourself. [link](https://github.com/HBClab/NiBetaSeries/invitations)

File: src/nibetaseries/cli/run.py (hunk 2 above)
Comment (tsalo) [reply src/nibetaseries/cli/run.py]: Okay, I added the check in `main`. Let me know if you'd prefer it implemented another way.

Comment (jdkent) [reply src/nibetaseries/cli/run.py]: I like the check you added in main, thanks!

**Comment by jdkent** [comment]
Great, work, I think this is pretty much ready. For documentation I would like an entry in [workflows.rst](https://github.com/HBClab/NiBetaSeries/blob/0bd10a139c2636fdfc3c3c69c625ff4e4c909e30/docs/workflows.rst#betaseries-workflow) showing how this method is different from normal `lss`.

**Comment by tsalo** [comment]
Should the description go in the paragraph below the Participant Workflow figure? What about in the Beta Series page?

**Comment by jdkent** [comment]
Good question. I'm imagining that under Beta Series Workflow section header, we will have several subsections; one for plain `lss`, one for `lss with fir`, and in a separate pull request add one for `lsa`. I can do LSA so you can have a model of what I'm thinking.

As for the Beta Series page, we can add a sentence or two under conceptual background. 

WDYT?

**Comment by tsalo** [comment]
Sounds good to me. Please let me know what you think of the changes.

**Pull request merged**